src/Microsoft.Extensions.DependencyInjection.Specification.Tests/DependencyInjectionSpecificationTests.Ordered.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection.Ordered;
+using Microsoft.Extensions.DependencyInjection.Specification.Fakes;
+using Xunit;
+
+namespace Microsoft.Extensions.DependencyInjection.Specification
+{
+    public abstract partial class DependencyInjectionSpecificationTests
+    {
+        [Fact]
+        public void AddOrdered_AllowsResolvingEmptyIOrdered()
+        {
+            // Arrange
+            var collection = new ServiceCollection();
+            collection.AddOrdered<IFakeService>();
+            var provider = CreateServiceProvider(collection);
+
+            // Act
+            var ordered = provider.GetService<IOrdered<IFakeService>>();
+
+            // Assert
+            Assert.Empty(ordered);
+        }
+
+        [Fact]
+        public void AddOrdered_AllowsResolvingAsIEnumerable()
+        {
+            // Arrange
+            var collection = new ServiceCollection();
+            collection.AddOrdered<IFakeService, FakeService>();
+            var provider = CreateServiceProvider(collection);
+
+            // Act
+            var ordered = provider.GetService<IEnumerable<IFakeService>>();
+
+            // Assert
+            Assert.IsType<FakeService>(ordered.Single());
+        }
+
+        [Fact]
+        public void AddOrdered_CachesInstances()
+        {
+            // Arrange
+            var collection = new ServiceCollection();
+            collection.AddOrdered<IFakeService, FakeService>();
+            var provider = CreateServiceProvider(collection);
+
+            // Act
+            var ordered = provider.GetService<IOrdered<IFakeService>>();
+            var array1 = ordered.ToArray();
+            var array2 = ordered.ToArray();
+
+            // Assert
+            Assert.Equal((IEnumerable<IFakeService>)array1, array2);
+        }
+
+        [Fact]
+        public void AddOrdered_SupportsTwoInstancesOfSameType()
+        {
+            // Arrange
+            var collection = new ServiceCollection();
+            collection.AddOrdered<IFakeService, FakeService>();
+            collection.AddOrdered<IFakeService, FakeService>();
+            var provider = CreateServiceProvider(collection);
+
+            // Act
+            var ordered = provider.GetService<IEnumerable<IFakeService>>();
```

Comment:
Yes, because in ServiceContainer we check for anything that inherits `IEnumerable`, I can limit this.